sklearn/model_selection/_split.py

```diff
@@ -1837,10 +1837,14 @@ def _validate_shuffle_split(n_samples, test_size, train_size):
     if test_size == "default":
         test_size = 0.1
 
-    if np.asarray(test_size).dtype.kind == 'f':
-        n_test = ceil(test_size * n_samples)
-    elif np.asarray(test_size).dtype.kind == 'i':
-        n_test = float(test_size)
+    if test_size is not None:
```

Comment:
why not put this into another elif instead of the else in the end? Would make the code slightly less complex imho.

Reply:
I considered that but thought it safer to use an else clause and this solution was similar to the structure in the _validate_shuffle_split_init above (see https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/model_selection/_split.py#L1794), which seemed pretty straight-forward.
I can change it to elif though, if you prefer.

Reply:
Why not avoid nesting by using
```
if test_size is None:
    pass
elif ...
```
?

Reply:
I changed it as @jnothman proposed and it actually goes quite nicely with the code that follows, i.e.
```python
if test_size is None:
...
if train_size is None:
...
if test_size is None:
...
```
Quite easy to spot the connection there, although this solution is a bit longer than @amueller's suggestion.
Let me know if you disagree or this needs more change.